test/tests/toolbar_toolbar.js

```diff
@@ -462,169 +449,344 @@ ariaTest('', exampleFile, 'toolbar-spinbutton-aria-valuemax', async (t) => {
+ariaTest(
+  'ARROW_RIGHT: Moves focus to the next control.',
+  exampleFile,
+  'toolbar-right-arrow',
+  async (t) => {
+    await t.context.session
+      .findElement(By.css(ex.itemSelectors.first))
+      .sendKeys(Key.ARROW_RIGHT);
+    await assertHasFocus(t, ex.itemSelectors.second);
+  }
+);
 
-    // Click on element to set focus
-    await clickAndWait(t, toolSelector);
+ariaTest(
+  'CLICK events on toolbar send focus to .item[tabindex="0"]',
+  exampleFile,
+  'toolbar-item-tabindex',
+  async (t) => {
+    let element = await t.context.session.findElement(By.css(ex.itemSelector));
+    await element.click();
+    await assertHasFocus(t, ex.itemSelector);
+  }
+);
 
-    // Send tab key to element
-    await t.context.session.findElement(By.css(toolSelector))
-      .sendKeys(Key.TAB);
+ariaTest(
+  'END: Moves focus to the last control.',
+  exampleFile,
+  'toolbar-home',
+  async (t) => {
+    await t.context.session
+      .findElement(By.css(ex.itemSelectors.first))
+      .sendKeys(Key.END);
+    await assertHasFocus(t, ex.itemSelectors.last);
+  }
+);
 
-    t.true(
-      await waitAndCheckFocus(t, ex.tabbableItemAfterToolbarSelector, index),
-      'Sending TAB to: ' + toolSelector + ' should move focus off toolbar'
+ariaTest(
+  'ESCAPE: Escape key hides any .popup-label',
+  exampleFile,
+  'toolbar-toggle-esc',
+  async (t) => {
+    await t.context.session.findElement(By.css(ex.itemSelectors.first)).click();
+    await sendKeyAndAssertSelectorIsHidden(
+      t,
+      Key.ESCAPE,
+      ex.itemSelectors.first,
+      '.popup-label.show'
     );
   }
-});
-
-ariaTest('key LEFT ARROW moves focus', exampleFile, 'toolbar-left-arrow', async (t) => {
-  // Put focus on the first item in the list
-  await clickAndWait(t, ex.allToolSelectors[0]);
-
-  let numTools = ex.allToolSelectors.length;
-  let toolSelector = ex.allToolSelectors[0];
-  let nextToolSelector = ex.allToolSelectors[numTools - 1];
-
-  // Send ARROW LEFT key to the first item
-  await t.context.session.findElement(By.css(toolSelector))
-    .sendKeys(Key.ARROW_LEFT);
-
-  // Focus should now be on last item
-  t.true(
-    await waitAndCheckFocus(t, nextToolSelector),
-    'Sending ARROW_RIGHT to tool "' + toolSelector + '" should move focus to "' +
-      nextToolSelector + '"'
-  );
+);
 
-  t.true(
-    await waitAndCheckTabindex(t, nextToolSelector),
-    'Sending ARROW_RIGHT to tool "' + toolSelector + '" should set tabindex on "' +
-      nextToolSelector + '"'
-  );
+ariaTest(
+  'HOME: Moves focus to the first control.',
+  exampleFile,
+  'toolbar-home',
+  async (t) => {
+    await t.context.session
+      .findElement(By.css(ex.itemSelectors.last))
+      .sendKeys(Key.HOME);
+    await assertHasFocus(t, ex.itemSelectors.first);
+  }
+);
 
+ariaTest(
+  'TAB: Moves focus into the toolbar, to the first menu item',
+  exampleFile,
+  'toolbar-tab',
+  async (t) => {
+    let tabTarget = ex.tabbaleItemBeforeToolbarSelector;
```

Comment:
Missing 'b' in tabbable:
```suggestion
    let tabTarget = ex.tabbableItemBeforeToolbarSelector;
```